### ISSUE TITLE
Issue with Azerbaijani Culture in .NET 8 Blazor WebAssembly Standalone App

### ISSUE THREAD

**Issue by xalidos** [dotnet/runtime#104717]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hello,

I am working on a .NET 8 Blazor WebAssembly Standalone App, and I have encountered an issue with setting the "az-Latn-AZ" (Azerbaijani) culture. While other cultures work correctly, Azerbaijani does not seem to be applied properly. Specifically, the ToLower function does not convert "Iı" to "ıı" as expected. Instead, it outputs "iı".

Here are the steps I've taken:

Added the culture settings to Program.cs:

```
CultureInfo.DefaultThreadCurrentCulture = new CultureInfo("az-Latn-AZ"); 
CultureInfo.DefaultThreadCurrentUICulture = new CultureInfo("az-Latn-AZ");
```

Added globalization settings to the project file (.csproj):

`<BlazorWebAssemblyLoadAllGlobalizationData>true</BlazorWebAssemblyLoadAllGlobalizationData>
`

Could anyone provide guidance on what might be going wrong or how to ensure the Azerbaijani culture settings are correctly applied in my Blazor WebAssembly app?

Thank you!

### Expected Behavior

`var toLower = "Iı".ToLower(); // Azerbaijani alphabet`

The output is "iı" instead of the expected "ıı".

### .NET Version

8.0.200

**Comment by maraf** [comment]
cc @ilonatommy

**Comment by tarekgh** [comment]
@ilonatommy could you please set the milestone when triaging the issues? Thanks!

**Comment by ilonatommy** [comment]
@xalidos, "az" is not included by default in WASM app internalization data, see: [available cultures](https://github.com/dotnet/icu/blob/42787a07ecba682549f520293bb5a6941bdccb50/icu-filters/icudt_wasm.json#L7).

`BlazorWebAssemblyLoadAllGlobalizationData` works only for Blazor WASM app (you say you are using pure WASM). It loads all cultures from the "available cultures" list mentioned above. An equivalent in WASM app is `<WasmIcuDataFileName>icudt_custom.dat</WasmIcuDataFileName>` [doc](https://github.com/dotnet/runtime/blob/ca0011a9698ec113535b710285bdda942ec5bd2b/docs/design/features/globalization-icu-wasm.md).

If you want to add any culture that is not added to "available cultures" cultures list (e.g. "az-Latn-AZ"), you should follow [custom ICU instruction](https://github.com/dotnet/runtime/blob/ca0011a9698ec113535b710285bdda942ec5bd2b/docs/design/features/globalization-icu-wasm.md#custom-icu) and use your own version of icu file, populating `WasmIcuDataFileName` with its name.

**Comment by xalidos** [comment]
@ilonatommy The issue here is not related to the Azerbaijani language not working. Azerbaijani language is functioning properly. It's just that the toLower() function is not working at the moment.

**Comment by ilonatommy** [comment]
@xalidos, I am sorry, I did not express myself clearly. I understand that for other methods Azerbaijani works correctly but in fact, it is just a coincidence, it seems it is failing back to a locale that just happens to have data to cover other use cases. It is sill missing the localization data specific to Azerbaijani. I created the custom file I mentioned above by adding: `"az_Latn_AZ"` to the "available cultures" I linked and built the ICU (`./build.sh /p:TargetOS=Browser /p:TargetArchitecture=wasm`).
![image](https://github.com/user-attachments/assets/ccd500d6-4978-4fbe-93ab-0909dc01140b)
[icudt_with_AZ.zip](https://github.com/user-attachments/files/16191546/icudt_with_AZ.zip)

I am attaching the file in case it was enough for you, if you will need more cultures / smaller ICU size, feel free to edit filters and build ICU on your own. To use it, remove `BlazorWebAssemblyLoadAllGlobalizationData` and add `<BlazorIcuDataFileName>$(MSBuildThisFileDirectory)icudt_with_AZ.dat</BlazorIcuDataFileName>` instead. Then, you'll see the result:
![image](https://github.com/user-attachments/assets/223f34fd-10d4-4efe-922d-384ce7496c36)
